nucliadb/nucliadb/common/maindb/pg.py

```diff
@@ -37,34 +37,39 @@
 class DataLayer:
     def __init__(self, connection: asyncpg.Connection):
         self.connection = connection
+        self.lock = asyncio.Lock()
```

Comment:
Why do we need a lock?

Reply:
Because a single connection can not have concurrent queries run against it